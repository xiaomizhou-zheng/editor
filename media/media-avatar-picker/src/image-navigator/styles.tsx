import { css, keyframes } from '@emotion/react';
import { borderRadius, gridSize } from '@atlaskit/theme/constants';
import { N200 } from '@atlaskit/theme/colors';
import { checkeredBg } from './images';

import { AVATAR_DIALOG_WIDTH } from '../avatar-picker-dialog/layout-const';

const spin = keyframes`
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
`;

export const imageBgStyles = css`
  position: absolute;
  top: 0;
  left: 0;
  width: 256px;
  height: 256px;
  background: url('${checkeredBg}');
  border-radius: ${borderRadius()};
`;

export const containerStyles = css`
  width: ${gridSize() * 32}px;
  box-sizing: border-box;
  *,
  *::before,
  *::after {
    box-sizing: border-box;
  }
  position: relative;
`;

export const sliderContainerStyles = css`
  align-items: center;
  justify-content: center;
  display: flex;
  flex-direction: row;
  margin-top: ${gridSize()}px;

  input {
    box-sizing: content-box;
    padding: 0;
  }
  background-color: #fff;
`;

export const fileInputStyles = css`
  display: none;
`;

export const imageUploaderStyles = css`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin: 0 10px 20px 10px;
`;

const droppingAnimation = css`
  border-color: #0e56c4;
  animation: ${spin} 8s linear infinite;
`;

export interface DragZoneProps {
  isDroppingFile: boolean;
  showBorder: boolean;
}

const getBorder = (showBorder: boolean) =>
  `${showBorder ? '2px dashed #d0d6d0' : 'none'}`;

const getDroppingAnimation = (isDroppingFile: boolean) =>
  isDroppingFile
    ? css`
        background-color: #ddecfe;

        &:after {
          ${droppingAnimation}
        }
      `
    : '';

export const dragZoneStyles = (props: DragZoneProps) => css`
  width: 200px;
  height: 200px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 15px;
  position: relative;
  border-radius: 100%;
  transition: background-color 0.3s cubic-bezier(0.215, 0.61, 0.355, 1);

  &::after {
    content: '';
    border: ${getBorder(props.showBorder)};
    border-radius: 100%;
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    transition: border-color 0.3s cubic-bezier(0.19, 1, 0.22, 1);
  }

  ${getDroppingAnimation(props.isDroppingFile)};
`;

dragZoneStyles.displayName = 'DragZone';

export const dragZoneImageStyles = css`
  width: 100px;
`;

const getWidth = (isFullSize: boolean) =>
  `${
    isFullSize
      ? `width:${AVATAR_DIALOG_WIDTH - gridSize() * 8}px`
      : 'width:auto'
  }`;

export interface DragZoneTextProps {
  isFullSize: boolean;
}

export const dragZoneTextStyles = (props: DragZoneTextProps) => css`
  text-align: center;
  color: ${N200};
  ${getWidth(props.isFullSize)};
`;

export const selectionBlockerStyles = css`
  position: fixed;
  left: 0;
  top: 0;
  right: 0;
  bottom: 0;
  background-color: transparent;
  user-select: none;
`;

export const paddedBreakStyles = css`
  margin-top: 10px !important;
  margin-bottom: 10px;
`;

export const sliderWrapperStyles = css`
  display: flex;
  align-items: center;
  width: 100%;

  .zoom_button svg {
    position: relative;
    left: -2px;
  }
`;
