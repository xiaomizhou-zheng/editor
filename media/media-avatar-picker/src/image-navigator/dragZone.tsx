/**@jsx jsx */
import { jsx } from '@emotion/react';
import { dragZoneStyles } from './styles';

export const DragZone = ({
  showBorder,
  isDroppingFile,
  children,
  ...props
}: any) => (
  <div
    css={dragZoneStyles({
      showBorder: showBorder,
      isDroppingFile: isDroppingFile,
    })}
    {...props}
  >
    {children}
  </div>
);
