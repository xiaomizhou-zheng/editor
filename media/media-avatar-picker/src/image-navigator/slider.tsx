/** @jsx jsx */
import { jsx } from '@emotion/react';
import { Component } from 'react';
import FieldRange from '@atlaskit/range';
import ScaleLargeIcon from '@atlaskit/icon/glyph/media-services/scale-large';
import ScaleSmallIcon from '@atlaskit/icon/glyph/media-services/scale-small';
import Button from '@atlaskit/button/custom-theme-button';
import { sliderWrapperStyles } from './styles';

export interface SliderProps {
  value: number;
  onChange: (value: number) => void;
}

export const defaultProps = {
  value: 0,
};

export class Slider extends Component<SliderProps, {}> {
  static defaultProps = defaultProps;

  render() {
    const { value, onChange } = this.props;
    return (
      <div css={sliderWrapperStyles}>
        <Button
          className="zoom_button zoom_button_small"
          iconAfter={<ScaleSmallIcon label="scale-small-icon" />}
          onClick={() => onChange(0)}
        />
        <FieldRange value={value} onChange={onChange} />
        <Button
          className="zoom_button zoom_button_large"
          iconAfter={<ScaleLargeIcon label="scale-large-icon" />}
          onClick={() => onChange(100)}
        />
      </div>
    );
  }
}
