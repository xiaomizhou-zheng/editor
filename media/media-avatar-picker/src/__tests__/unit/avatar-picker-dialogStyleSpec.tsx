import React from 'react';
import { render } from 'enzyme';
import { DragZone } from '../../image-navigator/dragZone';

describe('Avatar Picker Styles', () => {
  describe('image-navigator', () => {
    it('DragZone is dropping file', () => {
      const dragZone = render(
        <DragZone isDroppingFile={true} showBorder={true} />,
      );

      expect(dragZone).toMatchSnapshot();
    });

    it('DragZone is not dropping file', () => {
      const dragZone = render(
        <DragZone isDroppingFile={false} showBorder={true} />,
      );

      expect(dragZone).toMatchSnapshot();
    });
  });
});
