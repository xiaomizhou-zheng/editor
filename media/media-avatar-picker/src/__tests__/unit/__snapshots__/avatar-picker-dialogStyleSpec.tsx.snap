// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Picker Styles image-navigator DragZone is dropping file 1`] = `
@keyframes animation-0 {
  from {
    -webkit-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  to {
    -webkit-transform: rotate(360deg);
    -moz-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

.emotion-0 {
  width: 200px;
  height: 200px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 15px;
  position: relative;
  border-radius: 100%;
  -webkit-transition: background-color 0.3s cubic-bezier(0.215, 0.61, 0.355, 1);
  transition: background-color 0.3s cubic-bezier(0.215, 0.61, 0.355, 1);
  background-color: #ddecfe;
}

.emotion-0::after {
  content: '';
  border: 2px dashed #d0d6d0;
  border-radius: 100%;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  -webkit-transition: border-color 0.3s cubic-bezier(0.19, 1, 0.22, 1);
  transition: border-color 0.3s cubic-bezier(0.19, 1, 0.22, 1);
}

.emotion-0:after {
  border-color: #0e56c4;
  -webkit-animation: animation-0 8s linear infinite;
  animation: animation-0 8s linear infinite;
}

<div
  class="emotion-0"
/>
`;

exports[`Avatar Picker Styles image-navigator DragZone is not dropping file 1`] = `
.emotion-0 {
  width: 200px;
  height: 200px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 15px;
  position: relative;
  border-radius: 100%;
  -webkit-transition: background-color 0.3s cubic-bezier(0.215, 0.61, 0.355, 1);
  transition: background-color 0.3s cubic-bezier(0.215, 0.61, 0.355, 1);
}

.emotion-0::after {
  content: '';
  border: 2px dashed #d0d6d0;
  border-radius: 100%;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  -webkit-transition: border-color 0.3s cubic-bezier(0.19, 1, 0.22, 1);
  transition: border-color 0.3s cubic-bezier(0.19, 1, 0.22, 1);
}

<div
  class="emotion-0"
/>
`;
